### PR TITLE
5 controllers specs

### PR DESCRIPTION
**Pull request by OscarFMdev**
# PR description: 📄 

In this PR I:
- Tested the posts' and users' controllers 🧪 
- Added necessary gem for testing 💎 

### Thank you for reviewing my code! 🙌🏽

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    before(:each) { get user_posts_path(@user) }
+
+    it 'Gives the correct response status' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'Renders the correct template' do
+      expect(response).to render_template('index')
+    end
+  end
```
Comment (thecodechaser) [inline spec/requests/posts_spec.rb]: - [ ] You missed adding the test `If the response body includes correct placeholder text` for post-index action. Please make sure you add this test for post-index action.

Comment (thecodechaser) [inline spec/requests/posts_spec.rb]: - [ ] You missed adding all 3 tests for post-show action, Please make sure you add all tests for post-show action.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @OscarFMdev👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (OscarFMdev) [reply spec/requests/posts_spec.rb]: Ok, @thecodechaser thank you for letting me know!

Comment (OscarFMdev) [reply spec/requests/posts_spec.rb]: Thanks again @thecodechaser! It was giving me errors but then I remembered posts are nested from the user 😅

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Oscar, 
## To Highlight 🎉
All tests are passed. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by OscarFMdev** [comment]
Thank you @SajjadAhmad14!

**Pull request merged**